src/components/PearlThread/PearlThread.js

```diff
@@ -0,0 +1,119 @@
+import { h, Component } from 'preact';
+import classNames from 'classnames';
+import Icon from 'react-fontawesome';
+import { Colors } from './pearl-thread-style-constants';
+
+function onPearlThreadClick(onPearlClick, event) {
+    let target = event.target;
+
+    while (target) {
+        const pearlIndex = target.dataset && target.dataset.pearlIndex;
+
+        if (pearlIndex) {
+            onPearlClick(+pearlIndex);
+        }
+
+        target = target.parentNode;
+    }
+}
+
+const PearlThread = ({ items = [], activeIndex = 0, onPearlClick }) => {
+    const progressPercentage = (activeIndex / (items.length - 1)) * 100;
+    const isClickable = typeof onPearlClick === 'function';
+    const onContainerClick = isClickable ? onPearlThreadClick.bind(null, onPearlClick) : null;
+
+    return (
+        <div className="pearl-thread">
+            <div className="progress-bar" style={{ width: `${progressPercentage}%` }} ></div>
+
+            <div className="puzzle-progress-container"
+                onClick={onContainerClick}>
+                {items.map((item, index) => Pearl({
+                    caption: item.caption,
+                    index,
+                    activeIndex,
+                    isClickable,
+                    onPearlClick,
+                }))}
+            </div>
+
+
+            <style jsx>{`
+                .pearl-thread {
+                    min-height: 60px;
+                    padding: 0 20px;
+                }
+
+                .puzzle-progress-container {
+                    margin-top: 5px;
+                    display: flex;
+                    justify-content: space-between;
+                }
+
+                .progress-bar {
+                    background-color: ${Colors.PROGRESS_BAR_COLOR};
+                    width: 100%;
+                    height: 3px;
+                    transition: width 1s;
+                }
+            `}</style>
+        </div>
+    );
+}
+
+const Pearl = ({ caption, index, activeIndex, isClickable }) => {
+    let content;
+
+    if (index < activeIndex) {
+        content = <Icon name="check"/>;
+    } else if (index === activeIndex) {
+        content = caption;
+    } else {
+        content = index + 1;
+    }
+
+    const itemClass = classNames('pearl', {
+        'solved': index < activeIndex,
+        'current': index === activeIndex,
+        'future': index > activeIndex,
+        'clickable': isClickable,
+    });
+
+    return (
+        <div className={itemClass} data-pearl-index={index}>
+
+            { content }
+
+            <style jsx>{`
+                .pearl {
+                    width: 25px;
+                    height: 25px;
+                    line-height: 29px;
+                    text-align: center;
+                    color: ${Colors.UNSOLVED_PEARL_COLOR};
+                }
+
+                .pearl.solved {
+                    border: 2px solid #4c8580;
```

Comment:
Because of border Pearl's width is different when it's `solved` and `future`. As a result mark up jumps. Try to change selected index in Storybook

Reply:
I tried to position pearls with respect to current round label width in the latest commit.
It causes more jumps when selected index is changed, but gives more space for current round label.
Is it acceptable? Or is it better to use fixed positioning for pearls?

Reply:
Personally, I don't have a preference. Let's keep it as is for now.